x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,366 @@
+    const updateResults = await taskStore.bulkUpdate(taskUpdates, { validate: false });
+    for (const updateResult of updateResults) {
+      if (isOk(updateResult)) {
+        finalResults.push(updateResult.value);
+      } else {
+        const { id, type, error } = updateResult.error;
+
+        // this check is needed so error will be typed correctly for isConflictError
+        if (SavedObjectsErrorHelpers.isSavedObjectsClientError(error)) {
+          if (SavedObjectsErrorHelpers.isConflictError(error)) {
+            conflicts++;
+          } else {
+            logger.warn(
+              `Saved Object error updating task ${id}:${type} during claim: ${error.error}`,
+              logMeta
+            );
+            bulkErrors++;
+          }
+        } else {
+          logger.warn(`Error updating task ${id}:${type} during claim: ${error.message}`, logMeta);
+          bulkErrors++;
+        }
+      }
+    }
+  } catch (err) {
+    logger.warn(`Error updating tasks during claim: ${err}`, logMeta);
+  }
+
+  // separate update for removed tasks; shouldn't happen often, so unlikely
+  // a performance concern, and keeps the rest of the logic simpler
+  let removedCount = 0;
+  if (removedTasks.size > 0) {
+    const tasksToRemove = Array.from(removedTasks);
+    for (const task of tasksToRemove) {
+      task.status = TaskStatus.Unrecognized;
+    }
+
+    // don't worry too much about errors, we'll get them next time
+    try {
+      const removeResults = await taskStore.bulkUpdate(tasksToRemove, { validate: false });
+      for (const removeResult of removeResults) {
+        if (isOk(removeResult)) {
+          removedCount++;
+        } else {
+          const { id, type, error } = removeResult.error;
+          logger.warn(
+            `Error updating task ${id}:${type} to mark as unrecognized during claim: ${error.message}`,
+            logMeta
+          );
+        }
+      }
+    } catch (err) {
+      logger.warn(`Error updating tasks to mark as unrecognized during claim: ${err}`, logMeta);
+    }
+  }
+
+  const message = `task claimer claimed: ${finalResults.length}; stale: ${staleTasks.size}; conflicts: ${conflicts}; missing: ${missingTasks.size}; updateErrors: ${bulkErrors}; removed: ${removedCount};`;
+  logger.debug(message, logMeta);
+
+  const finalResult = {
+    stats: {
+      tasksUpdated: finalResults.length,
+      tasksConflicted: conflicts,
+      tasksClaimed: finalResults.length,
+    },
+    docs: finalResults,
+    timing: stopTaskTimer(),
+  };
+
+  for (const doc of finalResults) {
+    events$.next(asTaskClaimEvent(doc.id, asOk(doc), finalResult.timing));
+  }
+
+  return finalResult;
+}
+
+interface SearchAvailableTasksResponse {
+  docs: ConcreteTaskInstance[];
+  versionMap: Map<string, ConcreteTaskInstanceVersion>;
+}
+
+async function searchAvailableTasks({
+  definitions,
+  taskTypes,
+  removedTypes,
+  excludedTypes,
+  taskStore,
+  claimOwnershipUntil,
+  size,
+  taskMaxAttempts,
```

Comment:
we should be using this in the search though, to skip through tasks that have been tried too many times.  I'll see if I can add this, but if not, will add to the follow-up list ...

Reply:
hmmm ... I can see the default claimer passing this into the UBQ script, but it's not referencing it anywhere, near as I can tell.  Also, there is a function `taskWithLessThanMaxAttempts()`, which looks like would be used to generate DSL to check this, but it appears to be unused.  Guessing it used to be part of the query spec, then got moved to the UBQ script (and the function left in place, but no longer used), and then at some point we removed it from the UBQ script?  

I think this will require some more research :-) ...